src/hotspot/cpu/riscv/compressedKlass_riscv.cpp

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright (c) 2023, Red Hat, Inc. All rights reserved.
+ * Copyright (c) 2023, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ *
+ */
+
+#include "precompiled.hpp"
+#include "oops/compressedKlass.hpp"
+#include "utilities/globalDefinitions.hpp"
+
+char* CompressedKlassPointers::reserve_address_space_for_compressed_classes(size_t size, bool aslr, bool optimize_for_zero_base) {
+
+  char* result = nullptr;
+
+  // RiscV loads a 64-bit immediate in up to four separate steps, splitting it into four different sections
+  // (two 32-bit sections, each split into two subsections of 20/12 bits).
+  //
+  // 63 ....... 44 43 ... 32 31 ....... 12 11 ... 0
+  //       D           C          B           A
+  //
+  // A "good" base is, in this order:
+  // 1) only bits in A; this would be an address < 2KB, which is unrealistic on normal Linux boxes since
```

Comment:
@tstuefe : Hi Thomas, thanks for considering riscv. A small question: is the`2KB` in the code comment a typo? Seems that it should be `4KB` instead :-)

Reply:
Oh, you are right, thanks for catching that!